beacon_node/network/src/router/processor.rs

```diff
@@ -527,59 +527,52 @@ impl<T: BeaconChainTypes> Processor<T> {
-                            self.log,
-                            "Fork choice success";
-                            "location" => "block gossip"
-                        ),
-                        Err(e) => error!(
-                            self.log,
-                            "Fork choice failed";
-                            "error" => format!("{:?}", e),
-                            "location" => "block gossip"
-                        ),
-                    }
-                }
-                BlockProcessingOutcome::ParentUnknown { .. } => {
-                    // Inform the sync manager to find parents for this block
-                    // This should not occur. It should be checked by `should_forward_block`
-                    error!(self.log, "Block with unknown parent attempted to be processed";
-                            "peer_id" => format!("{:?}",peer_id));
-                    self.send_to_sync(SyncMessage::UnknownBlock(peer_id, block));
-                }
-                other => {
-                    warn!(
+                // TODO: It would be better if we can run this _after_ we publish the block to
+                // reduce block propagation latency.
+                //
+                // The `MessageHandler` would be the place to put this, however it doesn't seem
+                // to have a reference to the `BeaconChain`. I will leave this for future
+                // works.
+                match self.chain.fork_choice() {
+                    Ok(()) => trace!(
                         self.log,
-                        "Invalid gossip beacon block";
-                        "outcome" => format!("{:?}", other),
-                        "block root" => format!("{}", block.canonical_root()),
-                        "block slot" => block.slot()
-                    );
-                    trace!(
+                        "Fork choice success";
+                        "location" => "block gossip"
+                    ),
+                    Err(e) => error!(
                         self.log,
-                        "Invalid gossip beacon block ssz";
-                        "ssz" => format!("0x{}", hex::encode(block.as_ssz_bytes())),
-                    );
+                        "Fork choice failed";
+                        "error" => format!("{:?}", e),
+                        "location" => "block gossip"
+                    ),
                 }
-            },
-            Err(_) => {
-                // error is logged during the processing therefore no error is logged here
+            }
+            Err(BlockError::ParentUnknown { .. }) => {
+                // Inform the sync manager to find parents for this block
+                // This should not occur. It should be checked by `should_forward_block`
+                error!(
+                    self.log,
+                    "Block with unknown parent attempted to be processed";
+                    "peer_id" => format!("{:?}",peer_id)
```

Comment:
Just for future reference, `PeerId` now implements display, so I've been actively changing all these to `peer_id.to_string()` rather than using format. It will also resolve a clippy lint

Reply:
Sweet, I went and fixed the ones in this file.